builder/vsphere/supervisor/step_import_image.go

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+//go:generate packer-sdc struct-markdown
+//go:generate packer-sdc mapstructure-to-hcl2 -type ImportImageConfig
+
+package supervisor
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	imgregv1 "github.com/vmware-tanzu/image-registry-operator-api/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	ImportTargetKind       = "ContentLibrary"
+	ImportTargetAPIVersion = "imageregistry.vmware.com/v1alpha1"
+
+	DefaultWatchImageImportTimeoutSec = 600
+
+	ImportRequestDefaultNamePrefix    = "packer-vsphere-supervisor-import-req-"
+	StateKeyImageImportRequestCreated = "item_import_req_created"
+)
+
+var IsWatchingImageImport bool
+
+type ImportImageConfig struct {
+	// The name of the image import request.
+	// Defaults to `packer-vsphere-supervisor-import-req-<random-suffix>`.
+	ImportRequestName string `mapstructure:"import_request_name"`
+	// The timeout in seconds to wait for the image to be imported.
+	// Defaults to `600`.
+	WatchImportTimeoutSec int `mapstructure:"watch_import_timeout_sec"`
+}
+
+func (c *ImportImageConfig) Prepare() []error {
+	if c.WatchImportTimeoutSec == 0 {
+		c.WatchImportTimeoutSec = DefaultWatchImageImportTimeoutSec
+	}
+	if c.ImportRequestName == "" {
+		c.ImportRequestName = ImportRequestDefaultNamePrefix + uuid.NewString()[:5]
+	}
+
+	return nil
+}
+
+type StepImportImage struct {
+	ImportImageConfig  *ImportImageConfig
+	CreateSourceConfig *CreateSourceConfig
+
+	SourceURL, SourceSSLCertificate, TargetLocationName, TargetItemName, Namespace string
+	TargetItemType                                                                 imgregv1.ContentLibraryItemType
+	KeepInputArtifact                                                              bool
+	KubeWatchClient                                                                client.WithWatch
+}
+
+func (s *StepImportImage) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	logger := state.Get("logger").(*PackerLogger)
+
+	var err error
+	defer func() {
+		if err != nil {
+			state.Put("error", err)
+		}
+	}()
+
+	if err = s.initStep(state, logger); err != nil {
+		logger.Error("failed to initialize image import: %s", err)
+		return multistep.ActionHalt
+	}
+
+	// Skip importing if the import source URL or target location is not specified.
+	if s.SourceURL == "" || s.TargetLocationName == "" {
+		logger.Info("Skipping image import. Source URL or target location is not specified.")
+		return multistep.ActionContinue
+	}
```

Comment:
Can be removed since the step is added when these conditions are false. Plus, the `s.initStep()` can also guarantee that these configs are set.

Reply:
Keeping this one by updating other places.